fhir-model/src/main/java/com/ibm/fhir/model/util/FHIRUtil.java

```diff
@@ -131,9 +131,9 @@ public static String toString(Visitable visitable) {
     private static Pattern buildReferencePattern() {
         StringBuilder sb = new StringBuilder();
         sb.append("((http|https)://([A-Za-z0-9\\\\\\/\\.\\:\\%\\$\\-])*)?(");
-        sb.append(Arrays.asList(ResourceType.ValueSet.values()).stream()
-            .map(v -> v.value())
-            .collect(Collectors.joining("|")));
+        sb.append(ModelSupport.getResourceTypes(false).stream()
+                .map(t -> ModelSupport.getTypeName(t))
+                .collect(Collectors.joining("|")));
```

Comment:
Interesting... so we allowed Resource/1 and DomainResource/1 as valid references before. 
This is a good catch.

Reply:
yes, very astute